### ISSUE TITLE
use h264_rkmpp failed

### ISSUE THREAD

**Issue by nykh2010** [rockchip-linux/ffmpeg#2]
Impossible to convert between the formats supported by the filter 'ffplay_buffer' and the filter 'auto_scaler_0'.
location: libavfilter/avfiltergraph.c +606.
display failed. what is the reason?

**Comment by grkblood13** [comment]
I'm experiencing the same if not nearly identical problem. The following is a test file of an OTA ATSC broadcast called "out.ts": https://file.io/ZcjEOryb0vi7

Running the following command using the software encoder libx264 will result in a transcoded h264 file.

```
$ ffmpeg -c:v mpeg2video -f mpegts -i out.ts -f mp4 -c:v libx264 -c:a libfdk_aac -preset ultrafast -mbd rd -copyinkf -flags +ilme+ildct -fflags +genpts out.mp4 -y
ffmpeg version 589de66a0b Copyright (c) 2000-2018 the FFmpeg developers
  built with gcc 9.3.0 (GCC)
  configuration: --prefix=/usr/local --extra-libs='-lpthread -lm' --enable-hardcoded-tables --enable-gpl --enable-libass --enable-libfdk-aac --enable-libfreetype --enable-libmp3lame --enable-libopus --enable-libvorbis --enable-libvpx --enable-libx264 --enable-rkmpp --enable-version3 --enable-libdrm --enable-libx265 --enable-nonfree
  libavutil      56. 14.100 / 56. 14.100
  libavcodec     58. 18.100 / 58. 18.100
  libavformat    58. 12.100 / 58. 12.100
  libavdevice    58.  3.100 / 58.  3.100
  libavfilter     7. 16.100 /  7. 16.100
  libswscale      5.  1.100 /  5.  1.100
  libswresample   3.  1.100 /  3.  1.100
  libpostproc    55.  1.100 / 55.  1.100
[mpegts @ 0xaaaad00c8f00] PES packet size mismatch
Input #0, mpegts, from 'out.ts':                                                                                                                                                               
  Duration: 00:00:18.42, start: 29234.532200, bitrate: 5406 kb/s
    Stream #0:0[0x31]: Video: mpeg2video (Main), yuv420p(yuv420p) (tv, bt709, top first), 1920x1080 [SAR 1:1 DAR 16:9], Closed Captions, 29.97 fps, 29.97 tbr, 90k tbn, 59.94 tbc
    Stream #0:1[0x34]: Audio: ac3, 48000 Hz, 5.1(side), fltp, 384 kb/s
Stream mapping:
  Stream #0:0 -> #0:0 (mpeg2video (native) -> h264 (libx264))
  Stream #0:1 -> #0:1 (ac3 (native) -> aac (libfdk_aac))
Press [q] to stop, [?] for help
[libx264 @ 0xaaaad0143d10] using SAR=1/1
[libx264 @ 0xaaaad0143d10] using cpu capabilities: ARMv8 NEON
[libx264 @ 0xaaaad0143d10] profile Main, level 4.0, 4:2:0, 8-bit
[libx264 @ 0xaaaad0143d10] 264 - core 159 r2999M 296494a - H.264/MPEG-4 AVC codec - Copyleft 2003-2020 - http://www.videolan.org/x264.html - options: cabac=0 ref=1 deblock=0:0:0 analyse=0:0 me=dia subme=0 psy=1 psy_rd=1.00:0.00 mixed_ref=0 me_range=16 chroma_me=1 trellis=0 8x8dct=0 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=0 threads=6 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=tff bluray_compat=0 constrained_intra=0 bframes=0 weightp=0 keyint=250 keyint_min=25 scenecut=0 intra_refresh=0 rc=crf mbtree=0 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=0
Output #0, mp4, to 'out.mp4':e=       0kB time=-577014:32:22.77 bitrate=  -0.0kbits/s dup=19 drop=0 speed=N/A    
  Metadata:
    encoder         : Lavf58.12.100
    Stream #0:0: Video: h264 (libx264) (avc1 / 0x31637661), yuv420p(top coded first (swapped)), 1920x1080 [SAR 1:1 DAR 16:9], q=-1--1, 29.97 fps, 30k tbn, 29.97 tbc
    Metadata:
      encoder         : Lavc58.18.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 0/0/0 buffer size: 0 vbv_delay: -1
    Stream #0:1: Audio: aac (libfdk_aac) (mp4a / 0x6134706D), 48000 Hz, 5.1, s16, 488 kb/s
    Metadata:
      encoder         : Lavc58.18.100 libfdk_aac
[mpegts @ 0xaaaad00c8f00] PES packet size mismatch:00:17.95 bitrate=7476.8kbits/s dup=19 drop=0 speed=0.667x    
[ac3 @ 0xaaaad0144970] incomplete frame                                                                                                                                                        
frame=  552 fps= 20 q=-1.0 Lsize=   17568kB time=00:00:18.38 bitrate=7828.0kbits/s dup=19 drop=0 speed=0.668x                                                                                  
video:16491kB audio:1060kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.099193%
[libx264 @ 0xaaaad0143d10] frame I:3     Avg QP:18.00  size:296575
[libx264 @ 0xaaaad0143d10] frame P:549   Avg QP:20.17  size: 29137
[libx264 @ 0xaaaad0143d10] mb I  I16..4: 100.0%  0.0%  0.0%
[libx264 @ 0xaaaad0143d10] mb P  I16..4:  2.1%  0.0%  0.0%  P16..4: 47.6%  0.0%  0.0%  0.0%  0.0%    skip:50.4%
[libx264 @ 0xaaaad0143d10] field mbs: intra: 100.0% inter:100.0% skip:100.0%
[libx264 @ 0xaaaad0143d10] coded y,uvDC,uvAC intra: 54.4% 45.8% 16.0% inter: 18.4% 12.8% 0.6%
[libx264 @ 0xaaaad0143d10] i16 v,h,dc,p: 24% 54% 11% 10%
[libx264 @ 0xaaaad0143d10] i8c dc,h,v,p: 34% 40% 18%  7%
[libx264 @ 0xaaaad0143d10] kb/s:7334.32

```
Running the following command using the hardware encoder h264_rkmpp will result in an error message.

```
$ ffmpeg -c:v mpeg2video -f mpegts -i out.ts -f mp4 -c:v h264_rkmpp -c:a libfdk_aac -preset ultrafast -mbd rd -copyinkf -flags +ilme+ildct -fflags +genpts out.mp4 -y
ffmpeg version 589de66a0b Copyright (c) 2000-2018 the FFmpeg developers
  built with gcc 9.3.0 (GCC)
  configuration: --prefix=/usr/local --extra-libs='-lpthread -lm' --enable-hardcoded-tables --enable-gpl --enable-libass --enable-libfdk-aac --enable-libfreetype --enable-libmp3lame --enable-libopus --enable-libvorbis --enable-libvpx --enable-libx264 --enable-rkmpp --enable-version3 --enable-libdrm --enable-libx265 --enable-nonfree
  libavutil      56. 14.100 / 56. 14.100
  libavcodec     58. 18.100 / 58. 18.100
  libavformat    58. 12.100 / 58. 12.100
  libavdevice    58.  3.100 / 58.  3.100
  libavfilter     7. 16.100 /  7. 16.100
  libswscale      5.  1.100 /  5.  1.100
  libswresample   3.  1.100 /  3.  1.100
  libpostproc    55.  1.100 / 55.  1.100
[mpegts @ 0xaaaac67bff00] PES packet size mismatch
Input #0, mpegts, from 'out.ts':                                                                                                                                                               
  Duration: 00:00:18.42, start: 29234.532200, bitrate: 5406 kb/s
    Stream #0:0[0x31]: Video: mpeg2video (Main), yuv420p(yuv420p) (tv, bt709, top first), 1920x1080 [SAR 1:1 DAR 16:9], Closed Captions, 29.97 fps, 29.97 tbr, 90k tbn, 59.94 tbc
    Stream #0:1[0x34]: Audio: ac3, 48000 Hz, 5.1(side), fltp, 384 kb/s
Codec AVOption preset (Set the encoding preset (cf. x264 --fullhelp)) specified for output file #0 (out.mp4) has not been used for any stream. The most likely reason is either wrong type (e.g. a video option with no video streams) or that it is a private option of some encoder which was not actually used for any stream.                                                             
Stream mapping:                                                                                                                                                                                
  Stream #0:0 -> #0:0 (mpeg2video (native) -> h264 (h264_rkmpp))
  Stream #0:1 -> #0:1 (ac3 (native) -> aac (libfdk_aac))
Press [q] to stop, [?] for help
Impossible to convert between the formats supported by the filter 'Parsed_null_0' and the filter 'auto_scaler_0'
Error reinitializing filters!                                                                                                                                                                  
Failed to inject frame into filter network: Function not implemented                                                                                                                           
Error while processing the decoded data for stream #0:0                                                                                                                                        
Conversion failed!     

```

**Comment by JeffyCN** [comment]
Actually the mpp avdecoder do returning a drm_prime frame descriptor that ffmpeg understands, but it's the ffmpeg who doesn't support converting it(unlike gstreamer).

The newest version adds the ability to provide sw format yuv420sp(converted by rga hw module):
https://github.com/rockchip-linux/ffmpeg/commit/0ee87582de7a8a50da3c12e910a94859c9044ac6
